test/test_bq_billing_base.py

```diff
@@ -0,0 +1,885 @@
+
+        # order by 2 columns
+        self.assertEqual(
+            'ORDER BY cost ASC,day DESC',
+            BillingBaseTable._prepare_order_by_string(
+                {BillingColumn.COST: False, BillingColumn.DAY: True}
+            ),
+        )
+
+    def test_prepare_aggregation(self):
+        """Test _prepare_aggregation"""
+
+        query = BillingTotalCostQueryModel(
+            fields=[], start_date='2024-01-01', end_date='2024-01-01'
+        )
+
+        fields_selected, group_by = BillingBaseTable._prepare_aggregation(query)
+        # no fields selected so it is empty
+        self.assertEqual('', fields_selected)
+        # by default results are grouped by day
+        self.assertEqual('GROUP BY day', group_by)
+
+        # test when query is not grouped by
+        query = BillingTotalCostQueryModel(
+            fields=[BillingColumn.TOPIC],
+            start_date='2024-01-01',
+            end_date='2024-01-01',
+            group_by=False,
+        )
+
+        fields_selected, group_by = BillingBaseTable._prepare_aggregation(query)
+        # topic field is selected
+        self.assertEqual('topic', fields_selected)
+        # group by is switched off
+        self.assertEqual('', group_by)
+
+        # test when query is grouped by, but column can not be grouped by
+        # cost can not be grouped by, so it is not present in the result
+        query = BillingTotalCostQueryModel(
+            fields=[BillingColumn.TOPIC, BillingColumn.COST],
+            start_date='2024-01-01',
+            end_date='2024-01-01',
+            group_by=True,
+        )
+
+        fields_selected, group_by = BillingBaseTable._prepare_aggregation(query)
+        self.assertEqual('topic', fields_selected)
+        # always group by day and any field that can be grouped by
+        self.assertEqual('GROUP BY day,topic', group_by)
+
+    def test_prepare_labels_function(self):
+        """Test _prepare_labels_function"""
+
+        # test when there are no filters
+        query = BillingTotalCostQueryModel(
+            fields=[], start_date='2024-01-01', end_date='2024-01-01'
+        )
+
+        self.assertEqual(None, BillingBaseTable._prepare_labels_function(query))
+
+        # test when there are no labels in filters
+        query = BillingTotalCostQueryModel(
+            fields=[],
+            start_date='2024-01-01',
+            end_date='2024-01-01',
+            filters={BillingColumn.TOPIC: 'test_topic'},
+        )
+
+        self.assertEqual(None, BillingBaseTable._prepare_labels_function(query))
+
+        # test when there are labels in filters
+        query = BillingTotalCostQueryModel(
+            fields=[],
+            start_date='2024-01-01',
+            end_date='2024-01-01',
+            filters={BillingColumn.LABELS: {'test_key': 'test_value'}},
+        )
+
+        func_filter = BillingBaseTable._prepare_labels_function(query)
+
+        self.assertEqual('getLabelValue', func_filter.func_name)
```

Comment:
Are these tests still relevant with the label changes recently?

Reply:
Valid point, there are not used after we made change to the BQ table layout, 
let me remove all those labelFunction tests.